lib/archivers/zip/zip-archive-entry.js

```diff
@@ -199,14 +199,10 @@ ZipArchiveEntry.prototype.setTime = function(time) {
 };
 
 ZipArchiveEntry.prototype.setUnixMode = function(mode) {
-  mode &= ~constants.S_IFDIR;
-  var extattr = 0;
-
-  if (!this.isDirectory()) {
-    mode |= constants.S_IFREG;
-  }
+  mode &= ~constants.S_IFMT;
+  mode |= this.isDirectory() ? constants.S_IFDIR : constants.S_IFREG;
 
-  extattr &= ~this.getExternalAttributes();
```

Comment:
Ok, so I'm going to include 

var extattr = this.getExternalAttributes();
extattr |= (mode << constants.SHORT_SHIFT) | (this.isDirectory() ? constants.S_DOS_D : constants.S_DOS_A);

into my pull request, which will preserve all previously bits set.